src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -304,6 +311,17 @@ internal void SetCookieOption(Uri uri)
                 }
             }
 
+            private void SetClientCertificateOptions()
+            {
+                if (_clientCertProvider == null)
+                {
+                    return;
+                }
+                SetCurlOption(CURLoption.CURLOPT_SSLCERT, _clientCertProvider._certificateFile);
```

Comment:
It is read by the SSL library once (even before the Http call starts). The cert is needed during SSL negotiation.
But I do not know the requirement in case of a redirect, for instance. 
Apparently, it is a rare server that asks for client certificate during SSL negotiation. So testing it is a bit tricky. But I'll try and get the info.  

> and/or in the callback you're adding

If we are adding a callback, we do not need the file at all. Correct?